<a href="https://colab.research.google.com/github/talitalmeida/codes-java/blob/main/Lista1_CalcNumerico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Lista 1
Acadêmica: Talita Almeida



1.   Descrever em palavras uma aplicação ou uso do cálculo numérico.

Por meio do cálculo numérico é possível obter soluções numérica (aproximadas) de diversos problemas encontrados no mundo real, tais problemas podem ser solucionados por funções, matrizes, equações diferenciais etc. Por exemplo, em Ciência da Computação, o cálculo numérico é utilizado para o desenvolvimento de algoritmos na resolução de problemas complexos nas áreas de Inteligência Artificial, Machine Learning por meio da análise de dados, previsões de comportamentos.

2.   Escreva um algoritmo que converta números inteiros decimais em binários.



In [15]:
%%writefile DecToBin.java
import java.util.*;
public class DecToBin{
	public static void main(String[] args) {
		Scanner read = new Scanner(System.in);
			int numDecimal = read.nextInt();
			int numBinary = convertDecToBin(numDecimal);
			
			System.out.println(numBinary);
	}
	
	private static int convertDecToBin(int num_decimal) {
		String resto = "";
		for(int i = num_decimal; i > 0 ; i = i/2) {
			resto += String.valueOf(num_decimal%2);
			num_decimal /= 2;
		}
		
		String num_binary = new StringBuilder(resto).reverse().toString();
		
		return Integer.valueOf(num_binary);
	}
}

Writing DecToBin.java


In [16]:
%%shell 
javac DecToBin.java

In [17]:
%%shell
java DecToBin

23
10111


3. Escreva um algoritmo que converta números inteiros binários em decimais.



In [18]:
%%writefile BinToDec.java
import java.util.*;
public class BinToDec{
	public static void main(String[] args) {
		Scanner read = new Scanner(System.in);
			String numBinary = read.next();
			int numDecimal = convertBinToDec(numBinary);
			
			System.out.println(numDecimal);
	}

	private static int convertBinToDec(String bin) {
		int decimal = 0;
		
		for(int i = 0; i < bin.length(); i++) {
			int digito = Integer.valueOf(bin.substring(bin.length()-i-1, bin.length()-i));

			if(digito!=0) {
				decimal += (int)Math.pow(2, i);
			} 
		}

		return decimal;
	}
} 

Writing BinToDec.java


In [19]:
%%shell 
javac BinToDec.java

In [20]:
%%shell
java BinToDec

10111
23


4. Converta de decimal para binário ou binário para decimal

**a) 23**

quociente: 23/2 = 11\
resto: 1

quociente: 11/2 = 5\
resto 1

quociente: 5/2 = 2\
resto: 1

quociente: 2/2 = 1\
resto: 0

**Resultado: 10111**


**b) (1010)2**

2¹+2³ = **10**

**c) 0,1** \
0,1 * 2 = 0,2\
0,2 * 2 = 0,4\
0,4 * 2 = 0,8\
0,8 * 2 = 1,6\
0,6 * 2 = 1,2

**Resultado: 0,00011**

**d) 37**\
(1) 37/2 = 18\
(0) 18/2 = 9\
(1) 9/2 = 4\
(0) 4/2 = 2\
(0) 2/2 = (1)

**Resultado: 100101**

**e)0,1212** \
0,1212 * 2 = 0,2424\
0,2424 * 2 = 0,4848\
0,4848 * 2 = 0,9696\
0,9696 * 2 = 1,9392\
0,9392 * 2 = 1,8784\
0,8784 * 2 = 1,7568\
0,7568 * 2 = 1,5136\
0,5136 * 2 = 1,0272\
0,0272 * 2 = 0,0544

**Resultado: 0,000111110:**



**f)(101101)2** \
2⁰+2²+2³+2⁵ = **45** 

**g)(0,1101)2** \
(1 * 2⁻¹) + (1 * 2⁻²) + (1*2⁻⁴)

Resultado: 1/2 + 1/4 + 1/16 = **0,8125** 

5. Considere o sistema 𝐹(10,3, −4,4) normalizado. Represente neste sistema os números:

6. Seja um sistema de aritmética de ponto flutuante de quatro dígitos e base decimal. Dado os números: 𝑥 = 0,7237.10⁴; 𝑦 = 0,2145.10⁻³ e 𝑧 = 0,2585.10¹ . Efetue as seguintes
operações:

**a) x+y+z**\
x = 0,7237.10⁴\
y = 0,2145.10⁻³ = 0,00000002145⁴\
z = 0,2585.10¹ = 0,0002585⁴\

(0,7237 + 0,00000002145 + 0,0002585).10⁴ = \
**0,723958521.10⁴**

**b)x-y-z** \
x = 0,723700000000000.10⁴\
y = 0,2145.10⁻³ = 0,00000002145.10⁴\
z = 0,2585.10¹ = 0,00025850000.10⁴\

(0,72370000000 - 0,00000002145 - 0,00025850000).10⁴ = \
**0,723441479**

**c)x/y** \
(0,7237/0,2145).10⁴⁻³\
3,37389277.10¹ = **0,337389277.10²**

**d) (x.y)/z**\
((0,7237*0,2145).10⁴⁺²)/ 0,2585.10¹ \
0,15523365.10⁶/ 0,2585.10¹

(0,15523365/0,2585).10⁶⁻¹\
**0,600517021.10⁵**
